_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -836,7 +836,8 @@ mfxStatus ImplementationAvc::Init(mfxVideoParam * par)
 
     mfxStatus checkStatus = CheckVideoParam(m_video, m_caps, m_core->IsExternalFrameAllocator(), m_currentPlatform, m_currentVaType, *pMFXGTConfig, true);
     if (checkStatus == MFX_WRN_PARTIAL_ACCELERATION)
-        return MFX_WRN_PARTIAL_ACCELERATION;
+        // there is no sw fallback for AVC encoder, so reset the return value
+        return MFX_ERR_INVALID_VIDEO_PARAM;
```

Comment:
it is definitely wrong status to return. We should report UNSUPPORTED if no appropriate SW fallback

Reply:
AFAIR, it's `Query` function that is allowed to return UNSUPPORTED, `Init` have to return INVALID_VIDEO_PARAM